pkg/deployments/tasks/20230427-gauge-working-balance-helper/test/L1GaugeHelper.fork.ts

```diff
@@ -0,0 +1,171 @@
+    const stakeAmount = fp(100);
+
+    before('stake in gauge', async () => {
+      await lpToken.connect(lpTokenHolder).transfer(veBALHolder.address, stakeAmount);
+      await lpToken.connect(veBALHolder).approve(gauge.address, MAX_UINT256);
+
+      await gauge.connect(veBALHolder)['deposit(uint256)'](stakeAmount);
+    });
+
+    it('projected balance should equal current', async () => {
+      const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+        gauge.address,
+        veBALHolder.address
+      );
+
+      // Ensure we have equal balances (that are non-zero)
+      expect(projectedWorkingBalance).to.eq(currentWorkingBalance);
+      expect(projectedWorkingBalance).to.gt(0);
+    });
+  });
+
+  context('with veBAL', () => {
+    const TOKENLESS_PRODUCTION = 0.4;
+    const MAX_BALANCE_RATIO = 1 / TOKENLESS_PRODUCTION;
+
+    before('create veBAL whale', async () => {
+      bal80weth20Pool = await deployedAt('v2-pool-weighted/WeightedPool', VEBAL_POOL);
+      const poolId = await bal80weth20Pool.getPoolId();
+
+      await vault.connect(veBALHolder).joinPool(
+        poolId,
+        veBALHolder.address,
+        veBALHolder.address,
+        {
+          assets: [BAL, ZERO_ADDRESS],
+          maxAmountsIn: [0, VAULT_BOUNTY],
+          fromInternalBalance: false,
+          userData: WeightedPoolEncoder.joinExactTokensInForBPTOut([0, VAULT_BOUNTY], 0),
+        },
+        { value: VAULT_BOUNTY }
+      );
+
+      await bal80weth20Pool.connect(veBALHolder).approve(votingEscrow.address, MAX_UINT256);
+      const currentTime = await currentTimestamp();
+      await votingEscrow
+        .connect(veBALHolder)
+        .create_lock(await bal80weth20Pool.balanceOf(veBALHolder.address), currentTime.add(MONTH * 12));
+    });
+
+    it(`projected ratio should be greater than current by the maximum ratio (${MAX_BALANCE_RATIO})`, async () => {
+      const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+        gauge.address,
+        veBALHolder.address
+      );
+
+      expect(fromFp(projectedWorkingBalance) / fromFp(currentWorkingBalance)).to.eq(MAX_BALANCE_RATIO);
+    });
+
+    context('updates after checkpointing', () => {
+      before('checkpoint', async () => {
+        await gauge.connect(veBALHolder).user_checkpoint(veBALHolder.address);
+      });
+
+      it('projected balance should be close to or less than current', async () => {
+        const [currentWorkingBalance, projectedWorkingBalance] = await workingBalanceHelper.getWorkingBalances(
+          gauge.address,
+          veBALHolder.address
+        );
+
+        expect(projectedWorkingBalance).to.be.almostEqual(currentWorkingBalance);
+        expect(projectedWorkingBalance).to.be.gt(0);
+        expect(projectedWorkingBalance).to.be.lte(currentWorkingBalance);
+      });
+
+      it('current and projected ratios should now be equal', async () => {
```

Comment:
This is conceptually wrong, in that there's no reason why the rates would be equal. The rate could also be slightly lower. In this case, because we checkpoint and then immediately query it turns out both the balances and ratios are the same. A better test would be to check for them to be very close or equal here, then move forward e.g. one week, and then check that they are close but not the same.